### DISCUSSION TITLE
Notify new deployed versions

### DISCUSSION THREAD

**Discussion by fabidick22** [fluxcd/flux2#3978] [Q&A]
Hi team, I wonder if there is a possibility to notify only when a new version of a registry image is released. I'm using the Notification Controller to create an alert for certain events, but I can't find documentation to create an alert when a rollout event changes the version of the image I use in my deployment.

### Case of use:
The `Deployment` image is updated to version `v1.2.3` with the `ImageUpdateAutomation` resource and flux does its magic of reconciling the changes, at this point we would like to be able to notify the dev team that version `v1.2.3` was successfully released or not.

I understand that the Notification Controller can capture different [types of events](https://github.com/fluxcd/flux2/discussions/1183), but it cannot be filtered on this event because it is very generic, it can capture different types of changes on the `Deployment` resource.

So I would like to know some other event that I can capture or a workaround it to implement to be able to notify new deployed versions.

> **Note**: Probably this can be implemented with Flagger ([post-rollout](https://docs.flagger.app/usage/webhooks)) but I wanted to know if there is a possibility to implement it with flux before making a strong implementation of Flagger.

**Comment by makkes** [comment]
You can use your Kustomization as an [event source](https://fluxcd.io/flux/components/notification/alert/#event-sources).

**Reply by fabidick22** [reply]
Yes, I'm using this, but these events can be generated for different reasons, for example the change of the requested resources can generate a rollout event, I would like to be able to identify only the events where the image tag is changed.

**Reply by TOHUHM** [reply]
May I know if the events generated from kustomization source include image tag? I'm trying to get the deploy time too, then I can filter the event I want from 3rd part API

**Comment by TOHUHM** [comment]
@fabidick22  May I know if  you found a solution to get notification when new  release deployed?

**Reply by fabidick22** [reply]
We couldn't find a solution for this, so we opted for a more customized solution to our needs. 
The pain point is that after configuring events to notify a Slack channel, we noticed that we began receiving notifications when a deployment was restarted, when an attribute of the deployment manifest was changed (for example, a change in the requested resources), when a deployment was recreated, etc. It seems to me that currently, there is no way to filter only the event indicating that a deployment's image was updated. 
@stefanprodan If I'm wrong, please don't hesitate to correct me.

**Comment by stefanprodan** [comment]
The Flux Alert object supports custom metadata which is added to the alert message, so you can use the image automation marker and update the image tag like so:

```yaml
apiVersion: notification.toolkit.fluxcd.io/v1beta3
kind: Alert
metadata:
  name: apps
spec:
  eventSources:
    - kind: Kustomization
      name: apps
  eventMetadata:
    app1: image # {"$imagepolicy": "flux-system:app1"} 
```

When image automation updates the app image in the Deployment it will also update the Alert metadata. When the Alert will trigger for the Flux Kustomization rolling over the new app version, the notification will contain the latest image used.

**Reply by stefanprodan** [reply]
For wildcards usage my solution will not work. You'll need this RFC to be implemented: https://github.com/fluxcd/flux2/pull/4809

**Reply by TOHUHM** [reply]
I wonder what if the deployment yaml file and alert yaml file are not in the same Github repo?  why notification or some other component can't detect the new tag and return directly?
